internal/resources/cluster/resource_cluster.go

```diff
@@ -563,6 +588,55 @@ func withMetaUpdate(d *schema.ResourceData, cluster *clustermodel.VmwareTanzuMan
 	return true
 }
 
+func withTKGNodePoolUpdate(d *schema.ResourceData, nodepool *nodepoolmodel.VmwareTanzuManageV1alpha1ClusterNodepoolNodepool) bool {
+	switch {
+	case nodepool.Spec.TkgServiceVsphere != nil:
+		nodepools := d.Get(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, topologyKey, nodePoolKey)).([]interface{})[0].(map[string]interface{})
+		poolSpec := nodepools[SpecKey].([]interface{})[0].(map[string]interface{})
+		// poolInfo := nodepools["info"].([]interface{})[0].(map[string]interface{})
+		tkgsSpec := poolSpec[tkgServiceVsphereKey].([]interface{})[0].(map[string]interface{})
+
+		if d.HasChange(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, topologyKey, nodePoolKey)) {
+			incomingWorkerNodeCount := poolSpec[workerNodeCountKey].(string)
+
+			if incomingWorkerNodeCount != "" {
+				nodepool.Spec.WorkerNodeCount = incomingWorkerNodeCount
+			}
+
+			incomingTkgServiceVsphereClass := tkgsSpec[classKey].(string)
+
+			if incomingTkgServiceVsphereClass != "" {
+				nodepool.Spec.TkgServiceVsphere.Class = incomingTkgServiceVsphereClass
+			}
+
+			incomingTkgServiceVsphereStorageClass := tkgsSpec[storageClassKey].(string)
+
+			if incomingTkgServiceVsphereStorageClass != "" {
+				nodepool.Spec.TkgServiceVsphere.StorageClass = incomingTkgServiceVsphereStorageClass
+			}
+
+			log.Printf("[INFO] updating TKGs workload cluster nodepools")
+
+			return true
+		}
+	case nodepool.Spec.TkgVsphere != nil:
+		nodepools := d.Get(helper.GetFirstElementOf(SpecKey, tkgVsphereClusterKey, topologyKey, nodePoolKey)).([]interface{})[0].(map[string]interface{})
+		poolSpec := nodepools[SpecKey].([]interface{})[0].(map[string]interface{})
+
+		if d.HasChange(helper.GetFirstElementOf(SpecKey, tkgVsphereClusterKey, topologyKey, nodePoolKey)) {
+			incomingWorkerNodeCount := poolSpec[workerNodeCountKey].(string)
+
+			if incomingWorkerNodeCount != "" {
```

Comment:
Is there a reason to limit the update just to worker node count here?

Reply:
This is what the nodepool resource does. so rather than adding functionality I used what was already existing. If we want to add functionality I can do that. https://github.com/vmware/terraform-provider-tanzu-mission-control/blob/main/internal/resources/cluster/nodepools/resource_node_pool.go#L646-L660

Reply:
Yeah sound good, lets keep it the same as nodepool resource.